rdr_service/dao/consent_dao.py

```diff
@@ -13,65 +14,73 @@ class ConsentDao(BaseDao):
     def __init__(self):
         super(ConsentDao, self).__init__(ConsentFile)
 
-    def get_participants_with_consents_in_range(self, start_date, end_date=None) -> List[ParticipantSummary]:
-        with self.session() as session:
-            query = session.query(
-                ParticipantSummary
-            ).join(
-                Participant,
-                Participant.participantId == ParticipantSummary.participantId
-            ).filter(
-                ParticipantSummary.participantOrigin == 'vibrent',
-                Participant.isGhostId.isnot(True),
-                Participant.isTestParticipant.isnot(True),
+    @classmethod
+    def get_participants_with_consents_in_range(cls, session, start_date,
+                                                end_date=None) -> Collection[ParticipantSummary]:
+        query = session.query(
+            ParticipantSummary
+        ).join(
+            Participant,
+            Participant.participantId == ParticipantSummary.participantId
+        ).filter(
+            ParticipantSummary.participantOrigin == 'vibrent',
+            Participant.isGhostId.isnot(True),
+            Participant.isTestParticipant.isnot(True),
+            or_(
+                ParticipantSummary.email.is_(None),
+                ParticipantSummary.email.notlike('%@example.com')
+            )
+        )
+        if end_date is None:
+            query = query.filter(
                 or_(
-                    ParticipantSummary.email.is_(None),
-                    ParticipantSummary.email.notlike('%@example.com')
+                    ParticipantSummary.consentForStudyEnrollmentFirstYesAuthored >= start_date,
```

Comment:
If there's a bounded date range, I'm not sure I understand why we're limiting primary consent checks to FirstYes by that end date.    Does this still pick up validations for primary consent updates for Cohort 1?

Reply:
No, not at the moment. Validation for consent updates for cohort 1 still needs to be implemented.